src/classes/RecurringTask.ts

```diff
@@ -60,24 +62,66 @@ export class RecurringTask extends Task<RecurringTaskType> {
         this.endDate!,
         this.frequency!
       );
+      if (!this.verifyNoOverlap(recurringTask, taskArr)) {
+        throw new Error(`Unable to add recurring task ${this.name}: overlaps with existing task`);
+      }
       taskArr.push(recurringTask);
     }
   }
+  
+
+  private verifyNoOverlap(recurringTask: RecurringTask, taskArr: Task<string>[]): boolean {
```

Comment:
why can't we use verifyNoOverlap from the model instead of duplicating it, unless this is different somehow? Also wouldn't overlaps already be verified by the controller before calling appendTo?

Reply:
The controller doesnt check within creating recurrences itself so if it iterates and then there is an error in one of the inside recurrences (conflicting with other tasks), the controller cant catch it. So I thought its better to check inside the class itself

Reply:
This branch is too messed up so I created a fix-overlap branch, I confused myself too much here